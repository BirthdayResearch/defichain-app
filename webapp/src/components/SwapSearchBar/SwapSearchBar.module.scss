// App and Bootstrap variables required
@import '~bootstrap/scss/functions';
@import '~bootstrap/scss/variables';
@import '~bootstrap/scss/mixins';
@import '../../scss/variables';
@import '../../scss/bootstrap/variables';

.searchInput {
  border-right: 0;
}

.searchIndicator {
  width: 24px;
  height: 24px;
}
