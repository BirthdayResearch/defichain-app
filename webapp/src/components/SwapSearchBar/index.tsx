import React from 'react';
import { MdSearch } from 'react-icons/md';
import { Input, InputGroup, InputGroupAddon, InputGroupText } from 'reactstrap';
import styles from './SwapSearchBar.module.scss';

interface SwapSearchBarProps {
  searching: any;
  onChange: (e) => void;
  placeholder?: string;
}

const SwapSearchBar: React.FunctionComponent<SwapSearchBarProps> = (
  props: SwapSearchBarProps
) => {
  return (
    <InputGroup>
      <Input
        className={styles.searchInput}
        type='text'
        placeholder={props.placeholder}
        name='searchInput'
        id='searchInput'
        onChange={props.onChange}
      />
      <InputGroupAddon addonType='append'>
        <InputGroupText>
          <MdSearch className={styles.searchIndicator} />
        </InputGroupText>
      </InputGroupAddon>
    </InputGroup>
  );
};

export default SwapSearchBar;
