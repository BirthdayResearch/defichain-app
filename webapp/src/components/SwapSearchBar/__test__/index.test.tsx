import React from 'react';
import SwapSearchBar from '../index';
import { shallow } from 'enzyme';

describe('SwapSearchBar component', () => {
  it('should check for snapshot', () => {
    const props = {
      searching: 'searching',
      onChange: () => {},
      placeholder: 'placeholder',
    };
    const wrapper = shallow(<SwapSearchBar {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
