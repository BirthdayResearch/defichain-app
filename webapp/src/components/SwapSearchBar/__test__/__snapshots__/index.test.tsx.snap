// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SwapSearchBar component should check for snapshot 1`] = `
<InputGroup
  tag="div"
>
  <Input
    id="searchInput"
    name="searchInput"
    onChange={[Function]}
    placeholder="placeholder"
    type="text"
  />
  <InputGroupAddon
    addonType="append"
    tag="div"
  >
    <InputGroupText
      tag="span"
    >
      <MdSearch />
    </InputGroupText>
  </InputGroupAddon>
</InputGroup>
`;
